anime/views.py

```diff
@@ -45,31 +45,39 @@ def get_anime_by_name(name: str):
     else:
         abort(404)
 
-@animes_bp.route('/api/v1/animes/<int:id>', methods=['Put', 'Patch'])
-def anime(id: int):
+@animes_bp.route('/api/v1/animes/<int:id>', methods=['PUT', 'PATCH'])
+def update_anime(id: int):
     """Route updates attribute(s) of single anime in DB
 
-    :param name: Anime name to update its attribute(s)
-    :type name: str
+    :param id: Anime ID
+    :type id: str
     :return: Updated anime record
     :rtype: AnimeSchema
     """
 
-    try:
-        request_data = request.get_json()
-        db.session.query(Anime).filter(Anime.anime_id == id).update(
-            {**request_data}
-        )
-        db.session.commit()
-
-        anime = db.session.execute(
-                    db.select(animes_tab).filter_by(anime_id=id)).first()
+    if request.method == 'PUT':
         anime_schema = AnimeSchema()
+    else:
+        anime_schema = AnimeSchema(partial=True)
+
+    entry_dict = anime_schema.loads(request.data)
+
+    db_anime = db.session.query(Anime).filter(Anime.anime_id == id)
```

Comment:
if you use get here instead would you get a integrity exception you can catch with a flask handler?

Reply:
The issue is the get method can't being used, that using it in line 65 makes db_anime without update method, which  is used in line 76 to update the anime in the DB.